moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -283,14 +283,15 @@ bool plan_execution::PlanExecution::isRemainingPathValid(const ExecutableMotionP
     collision_detection::CollisionRequest req;
     req.group_name = t.getGroupName();
     req.pad_environment_collisions = false;
-    moveit::core::RobotState state = plan.planning_scene->getCurrentState();
+    moveit::core::RobotState start_state = plan.planning_scene->getCurrentState();
     std::map<std::string, const moveit::core::AttachedBody*> current_attached_objects, waypoint_attached_objects;
-    state.getAttachedBodies(current_attached_objects);
-    waypoint_attached_objects = plan_components_attached_objects_[path_segment.first];
+    start_state.getAttachedBodies(current_attached_objects);
+    if (plan_components_attached_objects_.size() > static_cast<size_t>(path_segment.first))
+      waypoint_attached_objects = plan_components_attached_objects_[path_segment.first];
     for (std::size_t i = std::max(path_segment.second - 1, 0); i < wpc; ++i)
     {
       collision_detection::CollisionResult res;
-      state = t.getWayPoint(i);
+      moveit::core::RobotState state = t.getWayPoint(i);
```

Comment:
I'd suggest declaring `state` outside of the loop for efficiency, like this:

```
moveit::core::RobotState state;
for (...)
  ...
    state = t.getWaypoint(i);
```

Reply:
Thanks for the suggestion @AndyZe. Would you mind re-approving the PR?